R/verb-joins.R

```diff
@@ -211,20 +207,23 @@ anti_join.tbl_lazy <- function(x, y, by = NULL, copy = FALSE,
     ...
   )
 
-  x$lazy_query <- lazy_query
   x
 }
 
-
-add_join <- function(x, y, type, by = NULL, sql_on = NULL, copy = FALSE,
-                     suffix = NULL,
-                     auto_index = FALSE,
-                     na_matches = "never",
-                     x_as = NULL,
-                     y_as = NULL,
+add_join <- function(x,
+                     y,
+                     type,
+                     by,
+                     sql_on,
+                     copy,
+                     suffix,
+                     auto_index,
+                     na_matches,
+                     x_as,
+                     y_as,
                      call = caller_env()) {
   if (!is.null(sql_on)) {
-    by <- list(x = character(0), y = character(0), on = sql(sql_on))
+    by <- list(x = character(0), y = character(0), on = unclass(sql_on))
```

Comment:
It would be great to implement `ident()`, `ident_q()` and `sql()` on top of vctrs at some point. They have some annoying issues that make it difficult to work with...